### PR TITLE
Fix nil deref when account is in bad/incomplete state

### PR DESCRIPTION
**Pull request by matthiashanel**
Signed-off-by: Matthias Hanel <mh@synadia.com>

This endpoint is for debugging, so we shouldn't rely on things not being nil.
Also include validation results and missing mapping

### CODE REVIEW

File: server/monitor.go [hunk 1]
```diff
@@ -2005,68 +2012,121 @@ func (s *Server) accountInfo(accName string) (*AccountInfo, error) {
...
-				Subject:      jwt.Subject(k),
-				Type:         jwt.Service,
-				TokenReq:     v.tokenReq,
-				ResponseType: jwt.ResponseType(v.respType.String()),
-			},
-			ApprovedAccounts: []string{},
-		}
-		for name := range v.approved {
-			e.ApprovedAccounts = append(e.ApprovedAccounts, name)
+		var e ExtExport
```
Comment (kozlovic) [inline server/monitor.go:2027]: Would this be easier to read?
```
e := ExtExport{
   Subject: jwt.Subject(k),
   Type: jwt.Service,
}
if v != nil {
  e.TokenReq = v.tokenReq
  e.ResponseType = ..
  ...
}
```
Basically create the export with the fields that don't depend on `v` being nil or not, then complete in the `if v != nil` section.

File: server/monitor.go [hunk 2]
```diff
@@ -2005,68 +2012,121 @@ func (s *Server) accountInfo(accName string) (*AccountInfo, error) {
...
+				ApprovedAccounts: []string{},
+			}
+			for name := range v.approved {
+				e.ApprovedAccounts = append(e.ApprovedAccounts, name)
+			}
 		}
 		exports = append(exports, e)
 	}
 	for k, v := range a.exports.streams {
+		token := false
```
Comment (kozlovic) [inline server/monitor.go:2053]: Similar to previous comment. Would create the ExtExport{} with common stuff then fill `if v != nil`

File: server/monitor.go [hunk 3]
```diff
@@ -2005,68 +2012,121 @@ func (s *Server) accountInfo(accName string) (*AccountInfo, error) {
...
+			}
 		}
 		exports = append(exports, e)
 	}
 	imports := []ExtImport{}
 	for _, v := range a.imports.streams {
-		imports = append(imports, ExtImport{
-			Import: jwt.Import{
+		invalid := true
+		var imp jwt.Import
```
Comment (kozlovic) [inline server/monitor.go:2077]: If v is nil, then imp will be an empty jwt.Import structure. Wajt doe this mean to add it to the list of imports? We will have it as `invalid` but nothing from the Import details.

File: server/monitor.go [hunk 4]
```diff
@@ -2005,68 +2012,121 @@ func (s *Server) accountInfo(accName string) (*AccountInfo, error) {
...
+		}
+		imports = append(imports, ExtImport{
+			Import:  imp,
+			Invalid: invalid,
 		})
 	}
 	for _, v := range a.imports.services {
-		imports = append(imports, ExtImport{
-			Import: jwt.Import{
+		invalid := true
```
Comment (kozlovic) [inline server/monitor.go:2092]: Same comment than above. What does it mean to add an ExtImport with an empty jwt.Import structure?

**Review by kozlovic — COMMENTED** [review]
Some code readability suggestions and really wonder what it means to add a ExtImport with an empty jwt.Import struct..

File: server/monitor.go (hunk 1 above)
Comment (matthiashanel) [reply server/monitor.go:2027]: fixed, essentially creating the empty export, overwriting in case of v != nil

File: server/monitor.go (hunk 2 above)
Comment (matthiashanel) [reply server/monitor.go:2053]: fixed, same as above

File: server/monitor.go (hunk 3 above)
Comment (matthiashanel) [reply server/monitor.go:2077]: the use case of /accountz is debugging, so right now I don't know what to expect.
The important bit would be to communicate that it's there but smth is wrong.
just that, smth is wrong invalid = true. no other details to be had.

File: server/monitor.go (hunk 4 above)
Comment (matthiashanel) [reply server/monitor.go:2092]: same as above

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**